### PR TITLE
Refactor assertions

### LINKED ISSUE #117: rename `ruleset` into `assertion`
We are going to use the terminology `assertion`, instead of `ruleset`

### PR DESCRIPTION
**Pull request by BillyWooo**
resolves #117

### CODE REVIEW

File: litentry/core/stf-task/receiver/src/stf_task_receiver.rs [hunk 1]
```diff
@@ -92,20 +92,35 @@ where
...
-					let result = lc_ruleset_build::build_ruleset(
-						request.who.clone(),
-						identity,
-						request.ruleset.clone(),
-					)
-					.map_err(|e| Error::RulesetError(format!("error verify ruleset: {:?}", e)));
+			RequestType::AssertionVerification(request) => {
+				match request.assertion {
+					Assertion::Assert1 => {},
+					Assertion::Assert2(guilt_id, user_id) => {
```
Comment (Kailai-Wang) [inline litentry/core/stf-task/receiver/src/stf_task_receiver.rs]: typo: `guild_id`

File: litentry/core/stf-task/sender/src/lib.rs [hunk 2]
```diff
@@ -97,9 +97,9 @@ impl From<Web3IdentityVerificationRequest> for RequestType {
 	}
 }
 
-impl From<RulesetVerificationRequest> for RequestType {
-	fn from(r: RulesetVerificationRequest) -> Self {
-		RequestType::RulesetVerification(r)
+impl From<AssertionVerificationRequest> for RequestType {
```
Comment (Kailai-Wang) [inline litentry/core/stf-task/sender/src/lib.rs]: Maybe `AssertionBuildRequest` to align with the crate name

File: litentry/primitives/src/assertion.rs [hunk 3]
```diff
@@ -28,18 +28,18 @@ type MaxStringLength = ConstU32<64>;
...
-	R6,
-	R7(Balance, u32), // (DOT_amount, year)
-	R8(u64),          // (tx_amount)
-	R9,
-	R10(Balance, u32), // (WBTC_amount, year)
-	R11(Balance, u32), // (ETH_amount, year)
-	R12(Balance, u32), // (LIT_amount, year)
-	R13(u32),          // (Karma_amount) - TODO: unsupported
+pub enum Assertion {
+	Assert1,
```
Comment (Kailai-Wang) [inline litentry/primitives/src/assertion.rs]: I still prefer shorter name like `A1`, as it's under `Assertion` enum already 😂

File: litentry/core/assertion-build/src/lib.rs [hunk 4]
```diff
@@ -36,23 +36,23 @@ use itc_rest_client::{error::Error as HttpError, RestPath};
...
-use itp_types::AccountId;
+pub mod assertion1;
+pub mod assertion2;
 
-use litentry_primitives::{Identity, Ruleset};
-
-pub mod discord;
-
-use crate::discord::ruleset2_verification;
+pub use crate::{assertion1::build_assertion1, assertion2::build_assertion2};
```
Comment (Kailai-Wang) [inline litentry/core/assertion-build/src/lib.rs]: My personal preference would be something like `a1::build`, and we only export the mod name but not the fn name.

So from outside it's `lc_assertion_build::a1::build()`, I think that's less verbose while still accurate

File: litentry/core/stf-task/receiver/src/stf_task_receiver.rs [hunk 5]
```diff
@@ -92,20 +92,35 @@ where
...
-				for identity in request.vec_identity {
-					let result = lc_ruleset_build::build_ruleset(
-						request.who.clone(),
-						identity,
-						request.ruleset.clone(),
-					)
-					.map_err(|e| Error::RulesetError(format!("error verify ruleset: {:?}", e)));
+			RequestType::AssertionVerification(request) => {
+				match request.assertion {
+					Assertion::Assert1 => {},
```
Comment (Kailai-Wang) [inline litentry/core/stf-task/receiver/src/stf_task_receiver.rs]: Why no processing?

File: app-libs/stf/src/trusted_call_litentry.rs [hunk 6]
```diff
@@ -118,51 +118,50 @@ impl TrustedCallSigned {
...
+	// 		}
+	// 	}
+
+	// 	if web2_cnt > 0 && web3_cnt > 0 {
+	// 		// TODO: generate_vc();
+	// 		Ok(())
+	// 	} else {
+	// 		Err(StfError::AssertionBuildFail)
+	// 	}
+	// }
```
Comment (Kailai-Wang) [inline app-libs/stf/src/trusted_call_litentry.rs]: Let's remove unneeded code (and a few other places)

**Review by Kailai-Wang — APPROVED** [review]
I have some different tastes regarding naming and some small adjustments.

Approved as they can be changed later

**Review by ziming-zung — APPROVED** [review]
Basically, it looks good to me.

**Pull request merged**